### DISCUSSION TITLE
RTNETLINK answers: Operation not permitted

### DISCUSSION THREAD

**Discussion by iksbob** [collin80/SavvyCAN#586] [Q&A] [answered]
I'm trying to snoop CAN messages on an automotive bus using a Microchip CAN analyzer, which is socketCAN compatible. Using the v208 binary on Ubuntu 20.4. So far I've installed libsocketcan (2, -dev and -doc via Synaptic), have not installed qt as it required signing up for an account and I got the impression it isn't needed to run the binary. 

Anyway, I get the title error in terminal when setting up the connection. QT SerialBus Devices > socketcan > can0

`qt.canbus: Can not disconnect an unconnected device.`
`RTNETLINK answers: Operation not permitted`
`qt.canbus.plugins.socketcan: Cannot apply parameter: 4 with value: 250000.`

The connection shows up in the Connected Devices list, with status "Connected". The activity lights on the adapter suggest CAN traffic (flickering mostly-on CAN RX light, flickering mostly-off CAN TX light, CAN ERR light on solid).

Selecting the device on the list and clicking the disconnect button generates:

`QSocketNotifier: Invalid socket 25 and type 'read', disabling...`

Do I need qt? Some other package? Maybe a permissions issue?

**Comment by collin80** [comment, marked as answer]
If you downloaded the Linux binary it should have had everything it needed already. However, there could be a few other things to check:

As the last error message showed, you can't set the speed of a socketcan link. The speed and other operational parameters are set from the command line before using it with SavvyCAN. Have you already initialized the socketcan interface? Here are the lines I run to do this:

sudo ip link set can0 type can bitrate 500000
sudo ifconfig can0 up

After that, it''s also a good idea to see if the command line socketcan tools can use the interface. If they can't then SavvyCAN can't either. So, try something like:

candump can0

And see if traffic starts streaming to the console. If so then SavvyCAN ought to be able to do so as well but if it can't then we can start troubleshooting from there.

Also, in a pinch, you can capture the output of candump and SavvyCAN is able to parse that data format. So, something like candump can0 > capture.log would let you capture some traffic then eventually load it into SavvyCAN. It's not an ideal approach but could allow for some traffic analysis while working out what is going on.

**Reply by iksbob** [reply]
That did it. I was getting pretty close reading through other googleings, but hadn't come across the bitrate limitation. I also installed can-utils, which it sounded like candump might be a part of?

Side note: the above error messages still pop up when connected. Is SavvyCAN trying to issue those same ip and ifconfig commands, but doesn't have permission? Seems like there aught to be a way for an app to operate CAN without root permissions. Just sayin.
